# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1390511690 18000
#      Thu Jan 23 16:14:50 2014 -0500
# Branch stable
# Node ID 50505bf88b8257300ab590c72f4245af14a4a7b1
# Parent  238aa26b30e3d164f38de539b9efafe17eab30d5
make octave-config and mkoctfile consistent with last shell script versions

* mkoctfile.in.cc, octave-config.in.cc: Update to match latest shell
script versions.

diff --git a/src/mkoctfile.in.cc b/src/mkoctfile.in.cc
--- a/src/mkoctfile.in.cc
+++ b/src/mkoctfile.in.cc
@@ -239,21 +239,24 @@ initialize (void)
   vars["FFTW3_LDFLAGS"] = get_variable ("FFTW3_LDFLAGS",
                                         %OCTAVE_CONF_FFTW3_LDFLAGS%);
   vars["FFTW3_LIBS"] = get_variable ("FFTW3_LIBS", %OCTAVE_CONF_FFTW3_LIBS%);
   vars["FFTW3F_LDFLAGS"] = get_variable ("FFTW3F_LDFLAGS",
                                          %OCTAVE_CONF_FFTW3F_LDFLAGS%);
   vars["FFTW3F_LIBS"] = get_variable ("FFTW3F_LIBS", %OCTAVE_CONF_FFTW3F_LIBS%);
   vars["LIBS"] = get_variable ("LIBS", %OCTAVE_CONF_LIBS%);
   vars["FLIBS"] = get_variable ("FLIBS", %OCTAVE_CONF_FLIBS%);
-  vars["OCTAVE_LINK_DEPS"] = get_variable ("FLIBS",
+  vars["OCTAVE_LINK_DEPS"] = get_variable ("OCTAVE_LINK_DEPS",
                                            %OCTAVE_CONF_OCTAVE_LINK_DEPS%);
-  vars["OCT_LINK_DEPS"] = get_variable ("FLIBS", %OCTAVE_CONF_OCT_LINK_DEPS%);
-  vars["FLIBS"] = get_variable ("FLIBS", %OCTAVE_CONF_FLIBS%);
-
+  vars["OCTAVE_LINK_OPTS"] = get_variable ("OCTAVE_LINK_OPTS",
+                                           %OCTAVE_CONF_OCTAVE_LINK_OPTS%);
+  vars["OCT_LINK_DEPS"] = get_variable ("OCT_LINK_DEPS",
+                                        %OCTAVE_CONF_OCT_LINK_DEPS%);
+  vars["OCT_LINK_OPTS"] = get_variable ("OCT_LINK_OPTS",
+                                        %OCTAVE_CONF_OCT_LINK_OPTS%);
   vars["LD_CXX"] = get_variable ("LD_CXX", %OCTAVE_CONF_MKOCTFILE_LD_CXX%);
   vars["LDFLAGS"] = get_variable ("LDFLAGS", %OCTAVE_CONF_LDFLAGS%);
   vars["LD_STATIC_FLAG"] = get_variable ("LD_STATIC_FLAG",
                                          %OCTAVE_CONF_LD_STATIC_FLAG%);
   vars["LFLAGS"] = get_variable ("LFLAGS", DEFAULT_LFLAGS);
   vars["F77_INTEGER8_FLAG"] = get_variable ("F77_INTEGER8_FLAG",
                                             %OCTAVE_CONF_F77_INTEGER_8_FLAG%);
 
@@ -291,16 +294,20 @@ static string help_msg =
 "  -DDEF                   Add -DDEF to compile commands.\n"
 "\n"
 "  -lLIB                   Add library LIB to link command.\n"
 "\n"
 "  -LDIR                   Add -LDIR to link command.\n"
 "\n"
 "  -M, --depend            Generate dependency files (.d) for C and C++\n"
 "                          source files.\n"
+#if ! defined (__WIN32__) || defined (_POSIX_VERSION)
+"\n"
+"  -pthread                Add -pthread to link command.\n"
+#endif
 "\n"
 "  -RDIR                   Add -RDIR to link command.\n"
 "\n"
 "  -Wl,...                 Pass flags though the linker like -Wl,-rpath=...\n"
 "\n"
 "  -W...                   Pass flags though the compiler like -Wa,OPTION.\n"
 "\n"
 "  -c, --compile           Compile, but do not link.\n"
@@ -309,39 +316,43 @@ static string help_msg =
 "                          (or .mex if --mex is specified) unless linking\n"
 "                          a stand-alone executable.\n"
 "\n"
 "  -g                      Enable debugging options for compilers.\n"
 "\n"
 "  -p VAR, --print VAR     Print configuration variable VAR.  Recognized\n"
 "                          variables are:\n"
 "\n"
-"                            ALL_CFLAGS                FFTW3F_LDFLAGS\n"
-"                            ALL_CXXFLAGS              FFTW3F_LIBS\n"
-"                            ALL_FFLAGS                FLIBS\n"
-"                            ALL_LDFLAGS               FPICFLAG\n"
-"                            AR                        INCFLAGS\n"
-"                            BLAS_LIBS                 LAPACK_LIBS\n"
-"                            CC                        LDFLAGS\n"
-"                            CFLAGS                    LD_CXX\n"
-"                            CPICFLAG                  LD_STATIC_FLAG\n"
-"                            CPPFLAGS                  LFLAGS\n"
-"                            CXX                       LIBOCTAVE\n"
-"                            CXXFLAGS                  LIBOCTINTERP\n"
-"                            CXXPICFLAG                LIBS\n"
-"                            DEPEND_EXTRA_SED_PATTERN  OCTAVE_LIBS\n"
-"                            DEPEND_FLAGS              OCTAVE_LINK_DEPS\n"
-"                            DL_LD                     OCT_LINK_DEPS\n"
-"                            DL_LDFLAGS                RANLIB\n"
-"                            EXEEXT                    RDYNAMIC_FLAG\n"
-"                            F77                       READLINE_LIBS\n"
-"                            F77_INTEGER_8_FLAG        SED\n"
-"                            FFLAGS                    XTRA_CFLAGS\n"
-"                            FFTW3_LDFLAGS             XTRA_CXXFLAGS\n"
-"                            FFTW3_LIBS\n"
+"                            ALL_CFLAGS                  INCFLAGS\n"
+"                            ALL_CXXFLAGS                INCLUDEDIR\n"
+"                            ALL_FFLAGS                  LAPACK_LIBS\n"
+"                            ALL_LDFLAGS                 LD_CXX\n"
+"                            AR                          LDFLAGS\n"
+"                            BLAS_LIBS                   LD_STATIC_FLAG\n"
+"                            CC                          LFLAGS\n"
+"                            CFLAGS                      LIBDIR\n"
+"                            CPICFLAG                    LIBOCTAVE\n"
+"                            CPPFLAGS                    LIBOCTINTERP\n"
+"                            CXX                         LIBS\n"
+"                            CXXFLAGS                    OCTAVE_HOME\n"
+"                            CXXPICFLAG                  OCTAVE_LIBS\n"
+"                            DEPEND_EXTRA_SED_PATTERN    OCTAVE_LINK_DEPS\n"
+"                            DEPEND_FLAGS                OCTAVE_LINK_OPTS\n"
+"                            DL_LD                       OCTAVE_PREFIX\n"
+"                            DL_LDFLAGS                  OCTINCLUDEDIR\n"
+"                            F77                         OCTLIBDIR\n"
+"                            F77_INTEGER8_FLAG           OCT_LINK_DEPS\n"
+"                            FFLAGS                      OCT_LINK_OPTS\n"
+"                            FFTW3F_LDFLAGS              RANLIB\n"
+"                            FFTW3F_LIBS                 RDYNAMIC_FLAG\n"
+"                            FFTW3_LDFLAGS               READLINE_LIBS\n"
+"                            FFTW3_LIBS                  SED\n"
+"                            FFTW_LIBS                   SPECIAL_MATH_LIB\n"
+"                            FLIBS                       XTRA_CFLAGS\n"
+"                            FPICFLAG                    XTRA_CXXFLAGS\n"
 "\n"
 "  --link-stand-alone      Link a stand-alone executable file.\n"
 "\n"
 "  --mex                   Assume we are creating a MEX file.  Set the\n"
 "                          default output extension to \".mex\".\n"
 "\n"
 "  -s, --strip             Strip output file.\n"
 "\n"
@@ -520,16 +531,22 @@ main (int argc, char **argv)
         {
           defs += (" " + arg);
         }
       else if (starts_with (arg, "-Wl,") || starts_with (arg, "-l")
                || starts_with (arg, "-L") || starts_with (arg, "-R"))
         {
           ldflags += (" " + arg);
         }
+#if ! defined (__WIN32__) || defined (_POSIX_VERSION)
+      else if (arg == "-pthread")
+        {
+          ldflags += (" " + arg);
+        }
+#endif
       else if (arg == "-M" || arg == "-depend" || arg == "--depend")
         {
           depend = true;
           compile = false;
         }
       else if (arg == "-o" || arg == "-output" || arg == "--output")
         {
           if (i < argc-1)
@@ -776,17 +793,17 @@ main (int argc, char **argv)
             {
               string cmd = vars["LD_CXX"] + " " + vars["CPPFLAGS"] + " "
                            + vars["ALL_CXXFLAGS"] + " " + vars["RDYNAMIC_FLAG"]
                            + " " + vars["ALL_LDFLAGS"] + " "
                            + pass_on_options + " " + output_option + " "
                            + objfiles + " " + libfiles + " "
                            + ldflags + " " + vars["LFLAGS"]
                            + " -loctinterp -loctave "
-                           + " " + vars["OCT_LINK_OPTS"]
+                           + " " + vars["OCTAVE_LINK_OPTS"]
                            + " " + vars["OCTAVE_LINK_DEPS"];
               result = run_command (cmd);
             }
           else
             {
               cerr << "mkoctfile: no way to link stand-alone executable file"
                    << endl;
               return 1;
diff --git a/src/octave-config.in.cc b/src/octave-config.in.cc
--- a/src/octave-config.in.cc
+++ b/src/octave-config.in.cc
@@ -48,33 +48,33 @@ static std::string help_msg =
 "                        expects to find locally installed .m files.\n"
 "\n"
 "  --oct-site-dir        Print the name of the directory where Octave\n"
 "                        expects to find locally installed .oct files.\n"
 "\n"
 "  -p VAR, --print VAR   Print the value of the given configuration\n"
 "                        variable VAR.  Recognized variables are:\n"
 "\n"
-"                          API_VERSION            LOCALAPIARCHLIBDIR\n"
-"                          CANONICAL_HOST_TYPE    LOCALAPIFCNFILEDIR\n"
-"                          DEFAULT_PAGER          LOCALAPIOCTFILEDIR\n"
+"                          API_VERSION            LOCALAPIOCTFILEDIR\n"
 "                          ARCHLIBDIR             LOCALARCHLIBDIR\n"
 "                          BINDIR                 LOCALFCNFILEDIR\n"
-"                          DATADIR                LOCALOCTFILEDIR\n"
-"                          DATAROOTDIR            LOCALSTARTUPFILEDIR\n"
-"                          EXEC_PREFIX            LOCALVERARCHLIBDIR\n"
-"                          FCNFILEDIR             LOCALVERFCNFILEDIR\n"
-"                          LOCALVEROCTFILEDIR     MAN1DIR\n"
-"                          IMAGEDIR               MANDIR\n"
-"                          MAN1EXT                OCTFILEDIR\n"
-"                          INCLUDEDIR             OCTINCLUDEDIR\n"
-"                          INFODIR                OCTLIBDIR\n"
-"                          INFOFILE               PREFIX\n"
-"                          LIBDIR                 STARTUPFILEDIR\n"
-"                          LIBEXECDIR             VERSION\n"
+"                          CANONICAL_HOST_TYPE    LOCALOCTFILEDIR\n"
+"                          DATADIR                LOCALSTARTUPFILEDIR\n"
+"                          DATAROOTDIR            LOCALVERARCHLIBDIR\n"
+"                          DEFAULT_PAGER          LOCALVERFCNFILEDIR\n"
+"                          EXEC_PREFIX            LOCALVEROCTFILEDIR\n"
+"                          FCNFILEDIR             MAN1DIR\n"
+"                          IMAGEDIR               MAN1EXT\n"
+"                          INCLUDEDIR             MANDIR\n"
+"                          INFODIR                OCTFILEDIR\n"
+"                          INFOFILE               OCTINCLUDEDIR\n"
+"                          LIBDIR                 OCTLIBDIR\n"
+"                          LIBEXECDIR             PREFIX\n"
+"                          LOCALAPIARCHLIBDIR     STARTUPFILEDIR\n"
+"                          LOCALAPIFCNFILEDIR     VERSION\n"
 "\n"
 "  -v, --version         Print the Octave version number.\n"
 "\n";
 
 static std::string
 substitute_prefix (const std::string& s, const std::string& prefix,
                    const std::string new_prefix)
 {
